wgpu-core/src/track/buffer.rs

```diff
@@ -701,15 +714,15 @@ unsafe fn insert<A: hub::HalApi>(
     log::trace!("\tbuf {index}: insert {new_start_state:?}..{new_end_state:?}");
 
     if let Some(&mut ref mut start_state) = start_states {
-        *start_state.get_unchecked_mut(index) = new_start_state;
+        unsafe { *start_state.get_unchecked_mut(index) = new_start_state };
     }
-    *current_states.get_unchecked_mut(index) = new_end_state;
+    unsafe { *current_states.get_unchecked_mut(index) = new_end_state };
 
-    let (epoch, ref_count) = metadata_provider.get_own(life_guard, index);
+    let (epoch, ref_count) = unsafe { metadata_provider.get_own(life_guard, index) };
 
     resource_metadata.owned.set(index, true);
-    *resource_metadata.epochs.get_unchecked_mut(index) = epoch;
-    *resource_metadata.ref_counts.get_unchecked_mut(index) = Some(ref_count);
+    unsafe { *resource_metadata.epochs.get_unchecked_mut(index) = epoch };
+    unsafe { *resource_metadata.ref_counts.get_unchecked_mut(index) = Some(ref_count) };
```

Comment:
This kind of thing seems fine to combine

Reply:
Combined in 0acce788.